File: code/mapreduce.ts
let shoppingList2 = ["catfish", "water", "tulips", "blue paint"];

shoppingList2
    .filter(item => item.startsWith("c"))
    .map(item => item.length)
    .reduce((acc, item) => acc + item)

shoppingList2
    .filter(item => item.startsWith("c"))
    .forEach(item => {
        console.log(item)
    })

File: code/mapreduce.kt
fun main13() {
    val shoppingList = arrayOf("catfish", "water", "tulips", "blue paint")
    shoppingList.filter { it.startsWith("c") }
        .map { it.length }
        .reduce { acc, i -> acc + i }

    shoppingList.filter { it.startsWith("c") }
        .forEach(::println)
}